.github/workflows/link-checker.yml

```diff
@@ -0,0 +1,93 @@
+name: Link Checker
+
+on:
+  repository_dispatch:
+  workflow_dispatch:
+  schedule: 
+    - cron: '0 0 1 * *' # Run at midnight on the first of every month
+    
+jobs: 
+  linkChecker:
+    name: Check and report broken links
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository  
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Corepack enable
+        run: corepack enable
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Get Token 
+        uses: actions/create-github-app-token@v1
+        id: get_workflow_token
+        with:
+          app-id: ${{ vars.APP_ID }}
+          private-key: ${{ secrets.PRIVATE_KEY }}
+
+      - name: Install Dependencies
+        run: pnpm install --frozen-lockfile
+        
+      - name: Serve App Locally
+        run: pnpm run dev &
+
+      - name: Wait for App to Start
+        run: sleep 20
+      
+      # This will restore the lychee cache
+      - name: Restore lychee cache
+        uses: actions/cache@v4
+        with:
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
+          restore-keys: cache-lychee-
+
+      # This will run the link checker on all markdown files in the pages directory      
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1
+        with:
+          args: --base https://json-schema.org  --verbose --no-progress --accept 200,204,429,403  './content/**/*.mdx' --cache --max-cache-age 1d  https://json-schema.org 
```

Comment:
I am not sure how the link checker workflow works, but why do we have json-shcema.org passed in the arguments? shouldn't it be tour.json-schema.org?  and if the workflow checks in the ./content/**/*.mdx files (which is what we want) then why are we passing the url of the website here?

Reply:
I see. I will have a look at it again.
I used the mentioned example workflow code to work upon. I will see what it is being used for and if we can remove it totally.

Reply:
I found that the link at the starting with `--base` is required if we are using relative links in markdown. 
The link at the end is added to make check it too in addition to all the links in `.mdx`. I think this is being used to check if our site itself is down which might be lead to links in mardown files to fail.
I will update the url with `https://tour.json-schema.org`

Reply:
Thanks @heysujal.